mail_tracking/models/mail_tracking_email.py

```diff
@@ -473,3 +473,26 @@ def event_process(self, request, post, metadata, event_type=None):
         # - return 'NONE' if this request is not for you
         # - return 'ERROR' if any error
         return "NONE"  # pragma: no cover
+
+    def _get_old_mail_tracking_email_domain(self, max_age_days):
+        target_write_date = fields.Datetime.subtract(
+            fields.Datetime.now(), days=max_age_days
+        )
+        return [("write_date", "<", target_write_date)]
+
+    @api.autovacuum
+    def _gc_mail_tracking_email(self, max_age_days=180, limit=5000):
+        enable_deletion = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("mail_tracking.enable_old_mail_tracking_email_deletion")
+        )
+
+        if not enable_deletion:
+            _logger.info("Mail tracking email deletion is disabled.")
+            return False
+
+        domain = self._get_old_mail_tracking_email_domain(max_age_days)
+        records_to_delete = self.search(domain, limit=limit)
+        _logger.debug("Deleting %s mail.tracking.email records", len(records_to_delete))
+        return records_to_delete.unlink()
```

Comment:
After testing this on a customer DB, it triggers the following error (linked to related fields):
```python
Traceback (most recent call last):
  File "/odoo/src/odoo/addons/base/models/ir_autovacuum.py", line 38, in _run_vacuum_cleaner
    func(model)
  File "/odoo/external-src/social/mail_tracking/models/mail_tracking_email.py", line 503, in _gc_mail_tracking_email
    records_to_delete.unlink()
  File "/odoo/external-src/connector/component_event/models/base.py", line 118, in unlink
    result = super(Base, self).unlink()
  File "/odoo/src/odoo/models.py", line 3501, in unlink
    self.flush()
  File "/odoo/src/odoo/models.py", line 5472, in flush
    self.recompute()
  File "/odoo/src/odoo/models.py", line 5931, in recompute
    process(field)
  File "/odoo/src/odoo/models.py", line 5918, in process
    field.recompute(existing)
  File "/odoo/src/odoo/fields.py", line 1178, in recompute
    self.compute_value(record)
  File "/odoo/src/odoo/fields.py", line 1198, in compute_value
    records._compute_field_value(self)
  File "/odoo/src/odoo/models.py", line 4079, in _compute_field_value
    odoo.fields.determine(field.compute, self)
  File "/odoo/src/odoo/fields.py", line 85, in determine
    return needle(records, *args)
  File "/odoo/src/odoo/fields.py", line 574, in _compute_related
    record[self.name] = self._process_related(value[self.related_field.name])
  File "/odoo/src/odoo/models.py", line 5702, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/odoo/src/odoo/fields.py", line 2495, in __get__
    return super().__get__(records, owner)
  File "/odoo/src/odoo/fields.py", line 1025, in __get__
    _("(Record: %s, User: %s)") % (record, env.uid),
odoo.exceptions.MissingError: Record does not exist or has been deleted.
```
I tried different things but only a raw `DELETE` query is working while being simple:
```suggestion
        records_to_delete = self.search(domain, limit=limit).exists()
        if records_to_delete:
            _logger.info("Deleting %s mail.tracking.email records", len(records_to_delete))
            self.flush()
            query = "DELETE FROM mail_tracking_email WHERE id IN %s"
            args = (tuple(records_to_delete.ids),)
            self.env.cr.execute(query, args)
            self.invalidate_cache()
```